mam/examples/ntru-key-exchange.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https:github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdio.h>
+#include "mam/api/api.h"
+#include "mam/ntru/ntru.h"
+
+int main() {
+  //Generating a keypair for the receiver api
+  mam_prng_t prng;
+  mam_ntru_sk_t ntru;
+  MAM_TRITS_DEF(nonce, 3 * 10);
+  MAM_TRITS_DEF(key, MAM_PRNG_SECRET_KEY_SIZE);
+  nonce = MAM_TRITS_INIT(nonce, 3 * 10);
+  key = MAM_TRITS_INIT(key, MAM_PRNG_SECRET_KEY_SIZE);
+  trits_from_str(key,
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC");
+  mam_prng_init(&prng, key);
```

Comment:
You may reuse `receiver_api.prng` for generating NTRU keypairs, that way you don't have to additionally generate/store secret `key` for the randomness generator.

Reply:
@semenov-vladyslav Do you mean to abandon all code before and including `mam_prng_init(&prng, key);` and just use `receiver_api.prng` in `ntru_sk_gen(&ntru, &prng, nonce);`? Do I still need to initialize `receiver_api.prng` or is this covered by `mam_api_init`?

Reply:
looked it up. `mam_api_init` does initialize the prng. done.